src/Microsoft.CodeQuality.Analyzers/Core/Maintainability/ReviewUnusedParameters.Fixer.cs

```diff
@@ -22,9 +30,133 @@ public sealed override FixAllProvider GetFixAllProvider()
 
         public sealed override Task RegisterCodeFixesAsync(CodeFixContext context)
         {
-            // Fixer not yet implemented.
+            Diagnostic diagnostic = context.Diagnostics.Single();
+
+            context.RegisterCodeFix(
+                new MyCodeAction(
+                    MicrosoftMaintainabilityAnalyzersResources.ReviewUnusedParametersMessage,
+                    async ct => await RemoveNodes(context.Document, diagnostic, ct).ConfigureAwait(false), diagnostic.Id),
+                diagnostic);
+
             return Task.CompletedTask;
+        }
+
+        protected abstract SyntaxNode GetParameterNode(SyntaxNode node);
+
+        private async Task<Solution> RemoveNodes(Document document, Diagnostic diagnostic, CancellationToken cancellationToken)
+        {
+            var solution = document.Project.Solution;
+            var pairs = await GetNodesToRemoveAsync(document, diagnostic, cancellationToken).ConfigureAwait(false);
+            foreach (var group in pairs.GroupBy(p => p.Key))
+            {
+                DocumentEditor editor = await DocumentEditor.CreateAsync(solution.GetDocument(group.Key), cancellationToken).ConfigureAwait(false);
+                // Start removing from bottom to top to keep spans of nodes that are removed later.
+                foreach (var value in group.OrderByDescending(v => v.Value.SpanStart))
+                {
+                    editor.RemoveNode(value.Value);
+                }
+
+                solution = solution.WithDocumentSyntaxRoot(group.Key, editor.GetChangedRoot());
+            }
+
+            return solution;
+        }
+
+        private ImmutableArray<IArgumentOperation>? GetOperationArguments(SyntaxNode node, SemanticModel semanticModel, CancellationToken cancellationToken)
+        {
+            while (node != null)
+            {
+                // For calls like A.B.C(0), it gets nulls for operations for first iterations. 
+                var operation = semanticModel.GetOperation(node, cancellationToken);
+
+                var arguments = (operation as IObjectCreationOperation)?.Arguments ?? (operation as IInvocationOperation)?.Arguments;
+
+                if (arguments.HasValue)
+                {
+                    return arguments.Value;
+                }
+
+                node = node.Parent;
```

Comment:
I am still slightly concerned with the approach. What if the original reference node is not using the method symbol in context of an invocation or an object creation, but it has some ancestor which is in context of an invocation or object creation? This algorithm will map the reference node to wrong invocation/object creation. Probably your original approach within a while loop was better as we at least know what syntax node kinds are we walking up.